sourcegraph/sourcegraph
# Getting started tour improvements

Closes https://github.com/sourcegraph/sourcegraph/issues/29890.

### Description
This PR:
- Adds new `disable-search-onboarding-tour` feature flag for disabling old search onboarding tour
   - Ex: `/search?feature-flag-key=disable-search-onboarding-tour&feature-flag-value=true` will disable locally
- Adds multiple feature flags override support through URL search parameters
    - Ex: `/search?feature-flag-key=my-feature1,my-feature-2&feature-flag-value=false,true`
- Getting started onboarding tour
   - Adds languages support + refactors to use zustand as a store
   - Adds support to show additional info dialog in search results
   - Minor content re-ordering/changes

### How to test
- `sg start web-standalone`
- https://sourcegraph.test:3443/search?feature-flag-key=disable-search-onboarding-tour,getting-started-tour&feature-flag-value=true,true
- Check that the old onboarding tour is not shown anymore (try in incognito mode, since there are other conditions to show or not for this onboarding tour)
- Check that getting started onboarding tour works with new languages support

> Note: for now only "Search diffs for added code" has lang-specific links, rest of items to be updated, waiting for the content.
